mutalyzer/grammar.py

```diff
@@ -83,9 +83,19 @@ class Grammar():
     # BNF: GeneName -> ([a-Z] | [0-9] | `-')+
     GeneName = Word(unicode(alphanums) + '-', min=1)
 
-    # BNF: GeneSymbol -> `(' Name (TransVar | ProtIso)? `)'
-    GeneSymbol = Suppress('(') + Group(GeneName('GeneSymbol') + \
-                 Optional(TransVar ^ ProtIso))('Gene') + Suppress(')')
+    # BNF: GeneProductID -> GeneName (TransVar | ProtIso)
+    GeneProductID = Group(GeneName('GeneSymbol') + \
+                          Optional(TransVar ^ ProtIso))('Gene')
+
+    # BNF: AccNoStem -> ([a-Z] Number `_')+
+    AccNoStem = NotAny('LRG_') + Combine(Word(unicode(alphas) + '_') + Number)
+
+    # BNF: AccNoFull -> AccNoStem `.' Number
+    AccNoFull = AccNoStem + Suppress('.') + Number
+
+    # BNF: GeneSymbol -> `(' (GeneProductID | AccNoFull) `)'
+    GeneSymbol = Suppress('(') + (GeneProductID ^ AccNoFull('AccNoTransVar')) \
+                 + Suppress(')')
```

Comment:
I'm not sure. Being agnostic here would make the code that handles the parsing output more complex or more inefficient I think. E.g. if the input is some unknown string, we'd have to treat it as a possible protein isoform, transcript variant and transcript accession number and then still fail if we can't find the string as any of these options.